### DISCUSSION TITLE
Getting the where-clause right on many-to-many relation queries

### DISCUSSION THREAD

**Discussion by riha** [prisma/prisma#12354] [Q&A] [answered]
I have a Prisma model for a Postgres database with a `User` that can belong to many `Organisations`, and organisations that can have many users. So I've created a typical many-to-many relation.

```
model Organization {
  organizationId Int      @id @default(autoincrement())
  slug           String   @unique
  name           String
  users          UserInOrganization[]
}

model User {
  userId        Int      @id @default(autoincrement())
  name          String
  organizations UserInOrganization[]
}

model UserInOrganization {
  fkOrganizationId Int
  fkUserId         Int
  organization     Organization @relation(fields: [fkOrganizationId], references: [organizationId], onDelete: Cascade)
  user             User         @relation(fields: [fkUserId], references: [userId], onDelete: Cascade)
  role             Role         @relation(fields: [fkRoleId], references: [roleId])

  @@id([fkOrganizationId, fkUserId])
}
```

Next, I need to get *all the organizations for a specific user and slug*. 

If I'd write it in raw SQL I'd do something like: 

```
select o.name, u."userId" from "Organization" as o 
join "UserInOrganization" as uio on o."organizationId" = uio."fkOrganizationId" 
join "User" as u on u."userId" = uio."fkUserId" 
where u."userId" = 1 and o.slug='slug'
````

But what would that be in Prisma? Doing something like below seems to give me ***all*** the organizations matching the slug (not filtering those the `where` on the `userId`). The `where` is just filtering the `users`. 

```
const organization = await prisma.organization.findUnique({
  where: { slug: slug },
  select: {
    name: true,
    users: { where: { user: { userId: userId } } },
  },
});
```

How can I get organizations that have the actual `userId` in the `UserInOrganization` table?

🙏🙏🙏

**Comment by austincrim** [comment, marked as answer]
Hey @riha 👋 ,

Could you help me understand how the `slug` plays into the query? Given your schema, every organization has a unique slug. A `findMany` query that filters on a specific slug will return **at most 1 organization**. Am I understanding that correctly?

If you weren't filtering on a specific `slug`, you could find all the organizations that are associated with a given `userId` like this:
```ts
let result = await prisma.organization.findMany({
  where: {
    users: {
      some: {
        user: {
          userId: userId
        }
      }
    }
  },
  select: {
    name: true
  }
})
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
